From: =?utf-8?q?Pablo_Correa_G=C3=B3mez?= <ablocorrea@hotmail.com>
Date: Mon, 14 Feb 2022 21:08:01 +0100
Subject: [PATCH] removable-media: Wrap GtkCheckButton label to fit narrow
 displays

GTK4 does not yet[1][2] allow to edit the wrap property of the associated
label. Until that happens and to allow fitting the panel in narrow displays,
split the label into its own object.

[1] https://gitlab.gnome.org/GNOME/gtk/-/issues/4698
[2] https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/4489
---
 panels/removable-media/cc-removable-media-panel.ui | 23 ++++++++++++++++------
 1 file changed, 17 insertions(+), 6 deletions(-)

Index: gnome-control-center/panels/removable-media/cc-removable-media-panel.ui
===================================================================
--- gnome-control-center.orig/panels/removable-media/cc-removable-media-panel.ui	2022-08-24 01:49:35.330000000 +0000
+++ gnome-control-center/panels/removable-media/cc-removable-media-panel.ui	2022-08-24 01:49:35.328000000 +0000
@@ -188,12 +188,23 @@
               </object>
             </child>
             <child>
-              <object class="GtkCheckButton" id="autorun_never_checkbutton">
-                <property name="valign">end</property>
-                <property name="vexpand">False</property>
-                <property name="use_underline">True</property>
-                <property name="label" translatable="yes">_Never prompt or start programs on media insertion</property>
-              </object>
+	      <object class="GtkBox">
+		<property name="orientation">horizontal</property>
+		<child>
+		  <object class="GtkCheckButton" id="autorun_never_checkbutton">
+                    <property name="valign">end</property>
+                    <property name="vexpand">False</property>
+		  </object>
+		</child>
+		<!-- Workaround for https://gitlab.gnome.org/GNOME/gtk/-/issues/4698 -->
+		<child>
+		  <object class="GtkLabel">
+		    <property name="use_underline">True</property>
+                    <property name="label" translatable="yes">_Never prompt or start programs on media insertion</property>
+		    <property name="wrap">True</property>
+		  </object>
+		</child>
+	      </object>
             </child>
           </object>
         </child>
